Issue59/contributing setup (#67)
lnovitz/jobseeker-analytics: CONTRIBUTING.md

This project uses **Google OAuth** for authentication. To run the app locally, you’ll need to configure your own Google API credentials.  

### Clone the repo
1. On Windows: We recommend that you use WSL2. [Installation instructions here](https://learn.microsoft.com/en-us/windows/wsl/). 
2. On Windows: start WSL 

> 3. In GIthub, fork the repository
> 3. Clone this fork  using ```git clone [URL copied from fork on github]```
> 4. ```cd``` into the repo you just cloned

Reviewer: ```suggestion
3. In Github, fork the repository
4. Clone this fork  using ```git clone [URL copied from fork on github]```
5. ```cd``` into the repo you just cloned
```

---

### Get a Google AI API key
1. Go to [https://aistudio.google.com/app/apikey](https://aistudio.google.com/app/apikey)
2. Click **Create and API Key**
3. Copy your API key and save it for later

---

### Create a Google OAuth App 
1. Go to the [Google Cloud Console](https://console.cloud.google.com/) and create a new project (or use an existing one).  
2. Navigate to **APIs & Services** → **Credentials**.  
3. If this is your first time creating credentials with this project, you will have to configure the OAuth consent screen.

> 4. On the OAuth Consent Screen page, scroll to "Test Users" and add yourself.

Reviewer: ```suggestion
4. On the OAuth Consent Screen page, scroll to "Test Users" and add your gmail address.
```

3. Click **Create Credentials** → **OAuth 2.0 Client IDs**.  
4. Set the application type to **Web Application**.  
5. Under "Authorized redirect URIs," add:  
   - https://jobseeker-analytics.onrender.com/login
   - http://localhost:8000/login
6. Copy the **Client ID** and **Client Secret** for later.  
7. Download and save your credentials locally to the parent folder for this repo in a file named ```credentials.json```

---

### Set Up Environment Variables

> 1. create a blank file called `.env`:  

Reviewer: ```suggestion
1. Copy `.env.example` to `.env`:
```
![Screenshot 2025-02-03 at 9 31 19 PM](https://github.com/user-attachments/assets/5bd5dcc6-b734-4fee-bee3-ff73d21f8b7a)

the existing `.env.example` file has the constants that need to be filled in already - easier to point users here: https://github.com/lnovitz/jobseeker-analytics/blob/main/.env.example
